test/account/paymaster/Paymaster.behavior.js

```diff
@@ -0,0 +1,96 @@
+const { ethers, entrypoint } = require('hardhat');
+const { expect } = require('chai');
+const time = require('@openzeppelin/contracts/test/helpers/time');
+const { encodeBatch, encodeMode, CALL_TYPE_BATCH } = require('@openzeppelin/contracts/test/helpers/erc7579');
+
+function shouldBehaveLikePaymaster({ postOp } = { postOp: false }) {
+  describe('entryPoint', function () {
+    it('should return the canonical entrypoint', async function () {
+      await expect(this.mock.entryPoint()).to.eventually.equal(entrypoint);
+    });
+  });
+
+  describe('validatePaymasterUserOp', function () {
+    beforeEach(async function () {
+      this.deposit = ethers.parseEther('1');
+      await this.mock.connect(this.depositor).deposit({ value: this.deposit });
+      this.userOp ??= {
+        paymaster: this.mock,
+      };
+    });
+
+    it('sponsors a user operation', async function () {
+      const userOp = { ...this.userOp };
+      userOp.callData = this.accountMock.interface.encodeFunctionData('execute', [
+        encodeMode({ callType: CALL_TYPE_BATCH }),
+        encodeBatch([
+          {
+            target: this.target.target,
+            data: this.target.interface.encodeFunctionData('mockFunction'),
+          },
+        ]),
+      ]);
+      const operation = await this.accountMock.createUserOp(this.userOp);
+      const userSignedUserOp = await this.signUserOp(operation);
+      const paymasterSignedUserOp = await this.paymasterSignUserOp(userSignedUserOp, 0, 0);
```

Comment:
Shouldn't the paymaster sign first, so that the paymaster signature can be placed in the userOp, and then the user signs that?